### DISCUSSION TITLE
Android14 emulator, after successfully installing the certificate, accessing website with ssl alerts  ssl/tls errors

### DISCUSSION THREAD

**Discussion by turinblueice** [mitmproxy/mitmproxy#6969] [Intercept Everything!]
#### Problem Description

On system Android 14  emulator，after successfully installing the certificate，the log outputs like below:

"Server TLS handshake failed. Certificate verify failed: unable to get local issuer certificate".

When I access the url with ssl like google.com, the error is :

Client TLS handshake failed. The client does not trust the proxy's certificate for www.google.com (OpenSSL Error([('SSL routines', '', 'ssl/tls alert certificate unknown')]))

That only appear on Android 14, not lower system.

The system certificate installed reffer to httptoolkit:   https://httptoolkit.com/blog/android-14-install-system-ca-certificate/

#### Steps to reproduce the behavior:
1.  Installed the cert on System14 in Emulator 
2. Start the mtimproxy service like: **mitmdump --set listen_port=6666  ssl_insecure=true**
3. Ensure the proxy can access the orignal server without ssl, but failed when access website with ssl.

![image](https://github.com/mitmproxy/mitmproxy/assets/3325170/08503d49-9939-4a6e-b386-f7f1647a0c13)

#### System Information
Mitmproxy: 10.3.0
Python:    3.10.0
OpenSSL:   OpenSSL 3.2.2 4 Jun 2024
Platform:  macOS-14.4-arm64-arm-64bit

**Comment by Prinzhorn** [comment]
Read about [certificate pinning](https://docs.mitmproxy.org/stable/concepts-certificates/#certificate-pinning)
